router/router.go

```diff
@@ -438,14 +474,16 @@ func (worker *workerT) workerProcess() {
 			batchDestination, ok := worker.rt.destinationsMap[parameters.DestinationID]
 			worker.rt.configSubscriberLock.RUnlock()
 			if !ok {
+				// we won't be using the throttling tokens anymore
+				go worker.rt.returnTokens(context.TODO(), tokensReturner)
```

Comment:
If this is the only place where we need to return tokens, then by simply moving this check before performing the throttling, we effectively eliminate the need for returning tokens altogether, correct?